Orleans/src/Microsoft.Orleans.Host.Abstractions/AzureStorageProvider.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Orleans.Host.Abstractions
+{
+    // this would go into the azure storage providers DLL
+    public static class AzureStorageProviderExtensions
+    {
+        public static void AddAzureStorage(this IStorageProvidersBuilder factory, string providerName, string connectionString)
```

Comment:
because this extendes the IStorageProvidersBuilder interface, not the concrete azure factory. In fact, as I mentioned in a comment, the factory itself might even go away, it's just an implementation detail.